### DISCUSSION TITLE
Button to update metadata list with textfield contents

### DISCUSSION THREAD

**Discussion by AfroSpartan** [mProjectsCode/obsidian-meta-bind-plugin#225] [Q&A] [answered]
I'm trying to create a button which adds the contents of a textfield to a list property called `skills` I current have the textfield bound to a text property called `newSkill` and I want to add the value from `newSkill` to the `skills` list and then clear the `newSkill` field.

I can't seem to figure out how to reference another property in the `updateMetadata` or `js` actions.

**Comment by Jacks-01** [comment]
Have you tried using a JS view field? There are some examples that are similar to what you are asking (I think) under [Advanced Use-Cases](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/advancedusecases/) in the documentation. They utilize [JS View Fields](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/viewfields/#creating-js-view-fields). 

I was able to dynamically update one bind value based off another bind-value when clicking a button using them. For example, setting the `current_hp` of my character back to full health (another bind value `total_hp`) I clicked a meta-bind button called "long rest".

**Reply by mProjectsCode** [reply, marked as answer]
Or you can use the JS button action to run a JS script where you first get the metadata you want and then update the other metadata based on that. (You can run button actions from JS using the plugins API)

**Comment by AfroSpartan** [comment]
I ended up using a JS file with the `metaedit` plugin

```JS
const metaedit = this.app.plugins.plugins["metaedit"].api;

let filePath = this.app.workspace.getActiveFile();
let skillObject = await metaedit.getPropertyValue("skills", filePath);
let newSkillName = await metaedit.getPropertyValue("newSkill", filePath);

skillObject[newSkillName] = "Neutral";

await metaedit.update("skills", skillObject, filePath);
await metaedit.update("newSkill", "", filePath);
```
